CentralityiOS/CentralityiOS/ViewControllers/CentralityHelpers.m

```diff
@@ -58,10 +67,23 @@ + (void) updateLabel:(UILabel*)label newText:(NSString*)newText isHidden:(BOOL)i
 }
 
 + (PFQuery*)queryForUsersCompletedTasks{
-    PFQuery *allUserTasks = [PFQuery queryWithClassName:kTaskClassName];
-    [allUserTasks whereKey:kByOwnerQueryKey equalTo:PFUser.currentUser];
-    [allUserTasks whereKeyExists:kByDateCompletedKey];
-    return allUserTasks;
+    PFQuery *completedTasks = [PFQuery queryWithClassName:kTaskClassName];
+    [completedTasks whereKey:kByOwnerQueryKey equalTo:PFUser.currentUser];
+    [completedTasks whereKeyExists:kByDateCompletedKey];
+    [completedTasks whereKey:kIsCompletedKey equalTo:@(YES)];
+    return completedTasks;
+}
+
+// Queries tasks due within 24 hour margin of error.
+// Not necessarily tasks due in the same day
+// When query called elsewhere, make sure to modify results to make them more accurate
++ (PFQuery*)queryForTasksRoughDueByDate{
```

Comment:
not necessary to change this, but since `PFQuery` is a part of the Parse library, you might want to think about how far you let it spread. If you do that and later on decide you want to swap out Parse for something else, you've just made your life a lot harder.

Unless there's a good performance reason not to, I would suggest extracting it into `NSArray *` as soon as possible.

Reply:
Scratch this - it looks like you're doing just that!